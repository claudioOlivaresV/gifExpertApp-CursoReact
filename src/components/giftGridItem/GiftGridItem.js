import React from 'react'
import { makeStyles } from '@material-ui/core/styles';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import IconButton from '@material-ui/core/IconButton';
import InfoIcon from '@material-ui/icons/Info';
import GridListTile from '@material-ui/core/GridListTile';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({

    icon: {
        color: 'rgba(255, 255, 255, 0.54)',
    },
}));

export const GiftGridItem = ({id, title, url:{ url }}) => {
        const classes = useStyles();

    
    return (
        <GridListTile key={id}>
        <img src={url} alt={title} />
        <GridListTileBar
          title={title}
          actionIcon={
            <IconButton aria-label={`info about ${title}`} className={classes.icon}>
              <InfoIcon />
            </IconButton>
          }
        />
      </GridListTile>
    )

}

GiftGridItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.object.isRequired,
}
