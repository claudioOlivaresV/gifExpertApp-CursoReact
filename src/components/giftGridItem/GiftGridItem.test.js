import React from 'react'
import "@testing-library/jest-dom/extend-expect"
import { mount, shallow } from 'enzyme'
import { GiftGridItem } from './GiftGridItem'

describe('Pruebas GiftGridItem', () => {

   
    
    const title = 'Un titulo';
    const id = 1;
    const url = {url: 'https://localhost'}
    
    const wrapper = shallow(<GiftGridItem id={id} title={title} url= {url} />)
    test('debe de mostrar el componente correctamente', () => {


        expect(wrapper).toMatchSnapshot();
    });
    test('debe de tener un titulo enviado', () => {
        
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url.url)
        expect(img.prop('alt')).toBe(title)
    });
  

});