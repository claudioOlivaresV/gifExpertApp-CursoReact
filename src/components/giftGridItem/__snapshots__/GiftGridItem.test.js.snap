// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GiftGridItem debe de mostrar el componente correctamente 1`] = `
<WithStyles(ForwardRef(GridListTile))
  key="1"
>
  <img
    alt="Un titulo"
    src="https://localhost"
  />
  <WithStyles(ForwardRef(GridListTileBar))
    actionIcon={
      <WithStyles(ForwardRef(IconButton))
        aria-label="info about Un titulo"
        className="makeStyles-icon-1"
      >
        <Memo />
      </WithStyles(ForwardRef(IconButton))>
    }
    title="Un titulo"
  />
</WithStyles(ForwardRef(GridListTile))>
`;
